File: js-pr/56/javascript/code-quality-performance/data_filter_18.js
// {fact rule=code-quality-performance@v1.0 defects=1}
function filterData(data, criteria) {
    let filtered = [];
    for (let item of data) {
        if (item.toLowerCase().includes(criteria.toLowerCase())) {
            filtered.push(item);
        }
    }
    return filtered;
}
// {/fact}

File: java-pr/49/java/performance/DataFilter18.java
// {fact rule=performance@v1.0 defects=1}
import java.util.*;
import java.util.stream.Collectors;
import java.io.*;

public class DataFilter18 {
    private Database database = new Database();
    
    public List<String> filterData(List<String> data, String criteria) {
        List<String> filtered = new ArrayList<>();
        for (String item : data) {
            if (item.toLowerCase().contains(criteria.toLowerCase())) {
                filtered.add(item);
            }
        }
        return filtered;
    }
    
    private static class Database {
        public User findUserById(int id) { return new User(); }
        public List<User> findAllUsers() { return new ArrayList<>(); }
    }
    
    private static class User {}
}
// {/fact}